tests/suites/test_suite_psa_its.function

```diff
@@ -66,6 +66,8 @@ static psa_status_t psa_its_set_wrap( psa_storage_uid_t uid,
                                       const void *p_data,
                                       psa_storage_create_flags_t create_flags )
 {
+    if( uid == 0 )
```

Comment:
Special-casing `uid==0` here doesn't make sense: it causes `set_UID_0_fail` to only validate the behavior of the test suite itself, and not the behavior of the library that it's testing. Either the test suite validates that the library implementation rejects uid=0, and then `library/psa_its_file.c` needs fixing; or the test suite doesn't validate this, and `set_UID_0_fail` is not useful.

Reply:
Yes, I agree, in psa_its_set would the check make much more sense. But I think we discussed that this would break of backward compatibility. Can we do it anyway?

Reply:
A user of Mbed TLS would have to go out of their way to use the functions from `psa_its_file.c` in their own code: these functions are not declared in the public headers of Mbed TLS. So I don't think there can be an expectation of backward compatibility about the behavior of `psa_its_file.c`.

There is a strong expectation of backward compatibility about the behavior of Mbed TLS features that rely on ITS. None of those features use 0 as the file UID: there's key storage (uses the key id, which can't be 0), entropy file (uses the uid `PSA_CRYPTO_ITS_RANDOM_SEED_UID` which is a nonzero constant), and dynamic SE driver storage (uses a range starting at `PSA_CRYPTO_SE_DRIVER_ITS_UID_BASE + 1`). So forbidding uid=0 in ITS won't affect the behavior of Mbed TLS.

Therefore changing `psa_its_set` to reject uid=0 will not affect applications using Mbed TLS, either directly or indirectly, unless the developer of the application has gone out of their way to use internal functions. If you use functions of a library that are evidently not part of its public inteface, you can't expect backward compatibility.

Now, whether we _should_ reject uid=0 in `psa_its_file`: I think we should, so that we don't accidentally write code that relies on uid=0 working. If we accidentally relied on uid=0 being supported, our code would work with `psa_its_file` (which is the only ITS implementation we test with) but it wouldn't work with other implementations. So I think that 1. `psa_its_set` from `psa_its_file.c` should reject uid=0, and 2. this behavior should be unit-tested.